notification/src/main/java/com/amazon/opendistroforelasticsearch/alerting/destination/client/DestinationMailClient.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.destination.client;
+
+import com.amazon.opendistroforelasticsearch.alerting.destination.message.BaseMessage;
+import com.amazon.opendistroforelasticsearch.alerting.destination.message.MailMessage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.Strings;
+import java.util.Properties;
+import javax.mail.*;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+/**
+ * This class handles the connections to the given Destination.
+ */
+public class DestinationMailClient {
+
+    private static final Logger logger = LogManager.getLogger(DestinationMailClient.class);
+    
+    public String execute(BaseMessage message) throws Exception {
+        if (message instanceof MailMessage) {
+            MailMessage mailMessage = (MailMessage) message;
+            Session session = null;
+
+            Properties prop = new Properties();
+            prop.put("mail.transport.protocol", "smtp");
+            prop.put("mail.smtp.host", mailMessage.getHost());
+            prop.put("mail.smtp.port", mailMessage.getPort());
+
+            if ( mailMessage.getUsername() != null && !mailMessage.getUsername().equals("".toCharArray())) {
+                prop.put("mail.smtp.auth", true);
+                session = Session.getInstance(prop, new javax.mail.Authenticator() {
+	    	        protected PasswordAuthentication getPasswordAuthentication() {
+                        try (
```

Comment:
Can add a `catch` to this `try` for better handling of the expected exception.

Also, the indentation for the `javax.mail.Authenticator` and the nested content is a little inconsistent. Reformatting that would help readability.